apps/ledger-live-desktop/src/renderer/modals/SignMessage/Body.tsx

```diff
@@ -23,25 +23,37 @@ type OwnProps = {
   data: Data;
 };
 type Props = OwnProps;
-const steps: Array<St> = [
-  {
-    id: "summary",
-    label: <Trans i18nKey="signmessage.steps.summary.title" />,
-    component: StepSummary,
-    footer: StepSummaryFooter,
-  },
-  {
-    id: "sign",
-    label: <Trans i18nKey="signmessage.steps.sign.title" />,
-    component: StepSign,
-    onBack: ({ transitionTo }: StepProps) => {
-      transitionTo("summary");
-    },
-  },
-];
+let steps: Array<St> = [];
+
 const Body = ({ onClose, data }: Props) => {
+  if (data.message.standard === "EIP712") {
+    steps = [
+      {
+        id: "sign",
+        label: "",
+        component: StepSign,
+      },
+    ];
+  } else {
+    steps = [
+      {
+        id: "summary",
+        label: <Trans i18nKey="signmessage.steps.summary.title" />,
+        component: StepSummary,
+        footer: StepSummaryFooter,
+      },
+      {
+        id: "sign",
+        label: <Trans i18nKey="signmessage.steps.sign.title" />,
+        component: StepSign,
+        onBack: ({ transitionTo }: StepProps) => {
+          transitionTo("summary");
+        },
+      },
+    ];
+  }
```

Comment:
We could put that in a useMemo or just create 2 different arrays outside the component for both flow